cmd/lotus-storage-miner/info.go

```diff
@@ -157,16 +164,47 @@ func infoCmdAct(cctx *cli.Context) error {
 	if !pow.HasMinPower {
 		fmt.Print("Below minimum power threshold, no blocks will be won")
 	} else {
-		expWinChance := float64(types.BigMul(qpercI, types.NewInt(build.BlocksPerEpoch)).Int64()) / 1000000
-		if expWinChance > 0 {
-			if expWinChance > 1 {
-				expWinChance = 1
-			}
-			winRate := time.Duration(float64(time.Second*time.Duration(build.BlockDelaySecs)) / expWinChance)
-			winPerDay := float64(time.Hour*24) / float64(winRate)
 
-			fmt.Print("Expected block win rate: ")
-			color.Blue("%.4f/day (every %s)", winPerDay, winRate.Truncate(time.Second))
+		winRatio := new(corebig.Rat).Mul(
+			new(corebig.Rat).SetFrac(
+				types.BigMul(pow.MinerPower.QualityAdjPower, types.NewInt(build.BlocksPerEpoch)).Int,
+				pow.TotalPower.QualityAdjPower.Int,
+			),
+			// decrease the rate ever-so-slightly to very roughly account for the multi-win poisson distribution
+			// FIXME - this is not a scientifically derived number... like at all
+			corebig.NewRat(99997, 100000),
+		)
+
+		if winRatioFloat, _ := winRatio.Float64(); winRatioFloat > 0 {
+
+			weekly, _ := new(corebig.Rat).Mul(
+				winRatio,
+				new(corebig.Rat).SetInt64(7*builtin.EpochsInDay),
+			).Float64()
+
+			avgDuration, _ := new(corebig.Rat).Mul(
+				new(corebig.Rat).SetInt64(builtin.EpochDurationSeconds),
+				new(corebig.Rat).Inv(winRatio),
+			).Float64()
+
+			fmt.Print("Projected average block win rate: ")
+			color.Blue(
+				"%.02f/week (every %s)",
+				weekly,
+				(time.Second * time.Duration(avgDuration)).Truncate(time.Second).String(),
+			)
+
+			// Geometric distribution calculated as described in https://en.wikipedia.org/wiki/Geometric_distribution#Probability_Outcomes_Examples
+			// https://www.wolframalpha.com/input/?i=t+%3E+0%3B+p+%3E+0%3B+p+%3C+1%3B+%281-%28p%29%29%5Et%3D%281-%28c%2F100%29%29%3B+solve+t
```

Comment:
Slack conversation led to further understanding here, will update in a few hours

Reply:
@Kubuxu I think we are done, tested it on nerpanet, the numbers match (but slightly better ;)

Thanks for all the handholding!